src/components/swap/SwapModalFooter.tsx

```diff
@@ -66,17 +96,16 @@ export default function SwapModalFooter({
   const [routerPreference] = useRouterPreference()
   const routes = isClassicTrade(trade) ? getRoutingDiagramEntries(trade) : undefined
   const theme = useTheme()
+  const [showMore, setShowMore] = useState(false)
 
   const lineItemProps = { trade, allowedSlippage, syncing: false }
 
   return (
     <>
+      <DropdownController open={showMore} onClick={() => setShowMore(!showMore)} />
       <DetailsContainer gap="md">
         <SwapLineItem {...lineItemProps} type={SwapLineItemType.EXCHANGE_RATE} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.PRICE_IMPACT} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.MAX_SLIPPAGE} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.MAXIMUM_INPUT} />
-        <SwapLineItem {...lineItemProps} type={SwapLineItemType.MINIMUM_OUTPUT} />
+        <AdvancedSwapDetails {...{ trade, allowedSlippage, open: showMore }} />
```

Comment:
i think you should make these into individual props for readability

Reply:
OOF i meant to refactor this ty

Reply:
embarassed

Reply:
ay lmao